File: 153. Find Minimum in Rotated Sorted Array/main.py
import sys
from typing import List


class Solution:
    def findMin(self, nums: List[int]) -> int:
        minVal = sys.maxsize

        for num in nums:
            if minVal > num:
                minVal = num

        return minVal


class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1

        while left < right:
            mid = (left + right) // 2
            if nums[mid] <= nums[right]:
                right = mid
            else:
                left = mid + 1

        return nums[left]

File: 153. Find Minimum in Rotated Sorted Array/main.c
#include <limits.h>
#include <stdio.h>

int findMin(int *nums, int numsSize) {
    int left = 0;
    int right = numsSize - 1;

    while (left < right) {
        int mid = (left + right) / 2;

        if (nums[mid] > nums[right]) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }

    return nums[left];
}

int findMin(int *nums, int numsSize) {
    int minVal = INT_MAX;

    for (int i = 0; i < numsSize; i++) {
        if (minVal > nums[i]) {
            minVal = nums[i];
        }
    }

    return minVal;
}